cmd/openshift-install/create.go

```diff
@@ -124,7 +125,9 @@ var (
 			// FIXME: add longer descriptions for our commands with examples for better UX.
 			// Long:  "",
 			PostRun: func(_ *cobra.Command, _ []string) {
-				ctx, cancel := context.WithCancel(context.Background())
+				// Setup a context that is canceled when the user presses Ctrl+C,
+				// or SIGTERM and SIGINT are received, this allows for a clean shutdown.
+				ctx, cancel := context.WithCancel(signals.SetupSignalHandler())
```

Comment:
We need the CAPI controllers to be shutdown, but this `ctx` is not passed through to those controllers. Do we need further signal handling [like this](https://github.com/openshift-cloud-team/installer/commit/2f8652a9e817715e12a144108ebe444d31917561#diff-1496fa801b4f49e1a37efad08459b950f8c54a5c281d559a1f056977e84ad172) for the controllers we're running as subprocesses?

Reply:
Reworked this a bit, the `signals.SetupSignalHandler` can only be used once, which makes sense because we're catching signals and there should only ever be one listener. I left a todo in the code to see if we can have the context to be passes through the commands to the asset chain eventually, but for now the current diff should do